tasks/build/build.js

```diff
@@ -50,19 +52,23 @@ const pack = () => {
   childProcess.execSync(`npm run pack`, { stdio: `inherit` })
 }
 
-function sha256File(path) {
-  let hash = createHash("sha256")
-  fs.createReadStream(path).pipe(hash)
-  return new Promise((resolve, reject) => {
-    hash.on("data", hash => resolve(hash.toString("hex")))
-  })
+const sha256 = async data => {
+  const hash = createHash(`sha256`).setEncoding(`base64`)
+
+  return data instanceof stream.Readable
```

Comment:
I think overloading the input to this function doesn't make it very readable.

Reply:
The purpose is to make the calling code to the function more readable.  The alternative would look like this:

```JavaScript
const inputsHash = await sha256Array([gaiaVersionHash, options])
const outputsHash = await sha256String(buildHashes)
const gaiaVersionHash = await sha256File(path.join(__dirname, `Gaia/COMMIT.sh`))
```

In the future there would need to be `sha256Object`, etc., for every new type that could be hashable.  It's better to make a more complex implementation in service of a simpler interface.

Reply:
The other way would be to only do sha256 of string to keep the interface clean and use `JSON.stringify` on arrays and objects to reduce those types to string. For files it is possible to just read the file content as string and hash this. Doing so makes it very clear what happens. In comparison to the developer needing to dive into a function to understand its functionality.

If you insist to keeping it this way, I don't want to block this from progressing.

Reply:
I'd like to understand your confusion better.  You're saying it's unclear how to use the function?

Would you make a similar argument about `JSON.stringify` and that it should be like this instead?

```JavaScript
JSON.stringifyArray
JSON.stringifyObject
JSON.stringifyString
JSON.stringifyNumber
JSON.stringifyBoolean
etc
```

Reply:
You are right. I would probably hate that. :)
Good explanation. Thank you.